webapp/src/components/sidebar/create_playbook_dropdown.tsx

```diff
@@ -0,0 +1,202 @@
+import {useIntl} from 'react-intl';
+import {getMyTeams} from 'mattermost-redux/selectors/entities/teams';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {Team} from 'mattermost-redux/types/teams';
+import {mdiClipboardPlayMultipleOutline, mdiImport} from '@mdi/js';
+import Icon from '@mdi/react';
+
+import {displayPlaybookCreateModal} from 'src/actions';
+import {importFile} from 'src/client';
+
+import Menu from '../widgets/menu/menu';
+import MenuItem from '../widgets/menu/menu_item';
+import MenuGroup from '../widgets/menu/menu_group';
+import MenuWrapper from '../widgets/menu/menu_wrapper';
+import {navigateToPluginUrl} from 'src/browser_routing';
+
+import {OVERLAY_DELAY} from 'src/constants';
+
+interface CreatePlaybookDropdownProps {
+    team_id: string;
+}
+
+const CreatePlaybookDropdown = (props: CreatePlaybookDropdownProps) => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const fileInputRef = useRef<HTMLInputElement | null>(null);
+    const teams = useSelector<GlobalState, Team[]>(getMyTeams);
+
+    const tooltip = (
+        <Tooltip id={'create_playbook_dropdown_tooltip'}>
+            {formatMessage({defaultMessage: 'Browse or create Playbooks and Runs'})}
+        </Tooltip>
+    );
+
+    const importUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e.target.files && e.target.files[0]) {
+            const file = e.target.files[0];
+
+            const teamId = props.team_id || teams[0].id;
+            const reader = new FileReader();
+            reader.onload = async (ev) => {
+                const {id} = await importFile(ev?.target?.result, teamId);
+                navigateToPluginUrl(`/playbooks/${id}/outline`);
+            };
+            reader.readAsArrayBuffer(file);
+        }
+    };
+
+    const renderDropdownItems = () => {
+        const browsePlaybooks = (
+            <MenuItem
+                id='browsePlaybooks'
+                show={true}
+                onClick={() => {
+                    navigateToPluginUrl('/playbooks');
+                }}
+                icon={<StyledIcon className='icon-globe'/>}
+                text={formatMessage({defaultMessage: 'Browse Playbooks'})}
+            />
+        );
+
+        const createPlaybook = (
+            <MenuItem
+                id='createPlaybook'
+                show={true}
+                onClick={() => dispatch(displayPlaybookCreateModal({}))}
+                icon={<StyledIcon className='icon-plus'/>}
+                text={formatMessage({defaultMessage: 'Create New Playbook'})}
+            />
+        );
+
+        const importPlaybook = (
+            <>
+                <MenuItem
+                    id='importPlaybook'
+                    show={true}
+                    onClick={() => {
+                        if (fileInputRef && fileInputRef.current) {
+                            fileInputRef.current.click();
+                        }
```

Comment:
Optional chaining again?
```suggestion
                    onClick={fileInputRef?.current?.click}
```